src/compositions/usePagination.ts

```diff
@@ -184,7 +184,14 @@ export function usePagination<
   }
 
   watch(fetchParametersGetter, () => {
-    page.value = 1
+    const newPageValue = 1
+    const manuallySetParameters = page.value === newPageValue
+
+    page.value = newPageValue
+
+    if (manuallySetParameters) {
+      setFetchSubscriptionParameters()
+    }
```

Comment:
I noticed that if the page value doesn't actually change the parameters would not get set because the page watcher wouldn't get triggered. This fixes that

Reply:
i'm surprised this is needed for something so simple but i don't have any other suggestions

Reply:
I believe its because vue doesn't rigger effects when assigning the same value and since `1 === 1` it doesn't trigger the watcher.